cumuluspeer/peer.go

```diff
@@ -0,0 +1,145 @@
+package cumuluspeer
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	host "github.com/libp2p/go-libp2p-host"
+	peer "github.com/libp2p/go-libp2p-peer"
+	net "github.com/libp2p/go-libp2p-net"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	swarm "github.com/libp2p/go-libp2p-swarm"
+	ma "github.com/multiformats/go-multiaddr"
+	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
+)
+
+const (
+	DefaultPort 	= 8765 // Cumulus peers communicate over this TCP port
+	CumulusProtocol	= "/cumulate/0.0.1" // Cumulus communication protocol
+	DefaultIP		= "127.0.0.1" // Default Host IP address if none given
+)
+
+// A basic StreamHandler for a very basic host.
+// This should be passed as the second argument to SetStreamHandler().
+// This is called when a remote peer opens a new stream with the host that
+// SetStreamHandler() is called on.
+// We may want to implement another type of StreamHandler in the future.
+func BasicStreamHandler(s net.Stream) {
+	log.Info("Setting basic stream handler.")
+	defer s.Close()
+	doCumulate(s)
+}
+
+// Communicate with peers.
+// TODO: Update this to do something useful. For now it just reads from the
+// stream and writes back what it read.
+func doCumulate(s net.Stream) {
+	buf := bufio.NewReader(s)
+	str, err := buf.ReadString('\n')
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	log.Info("Read: %s", str)
+	_, err = s.Write([]byte(str))
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	log.Info("Done now. Bye!")
+}
+
+// Create a Cumulus host.
+// This may throw an error if we fail to create a key pair, a pid, or a new
+// multiaddress.
+func MakeBasicHost(ip string, port int) (host.Host, pstore.Peerstore, error) {
```

Comment:
You should wrap `host.Host` with your own struct `peer.Peer` and return that here. Also use the pattern `NewPeer` rather than `MakePeer`